packages/library/src/commonMain/kotlin/io/realm/Realm.kt

```diff
@@ -112,9 +119,21 @@ class Realm private constructor(configuration: RealmConfiguration, dbPointer: Na
      *  this constructor should be the primary way to open Realms (as you only need
      *  to do it once pr. app).
      */
+    // FIXME This is not updated to the frozen architecture. Which way to go? Towards Realm.open or
```

Comment:
I think this should be the default way of opening a Realm similar to `Realm.getInstance()`

IMO we should make `Realm.open()` a suspend function and rework it to behave like `Realm.getInstanceAsync()`

A quick fix would be to make `Realm.open()` internal though.

Reply:
I added the existing workaround for obtaining the frozen realm from a live one and closing it into the `init`-clause, so `Realm(configuration)` does the right thing. We could basically just eliminate `Realm.open` for now, if we deem the fix ok, but only updated the `RealmTests` to use the constructor directly for now.